plugins/modules/mount.py

```diff
@@ -847,7 +944,8 @@ def main():
             # A non-working fstab entry may break the system at the reboot,
             # so undo all the changes if possible.
             try:
-                write_fstab(module, backup_lines, args['fstab'])
+                if state != 'ephemeral':
+                    write_fstab(module, backup_lines, args['fstab'])
```

Comment:
This piece of code is the restoration of the *fstab* file to its previous state when the module fails. When *state* is `ephemeral`, the *fstab* file is never modified, so this step must be skipped.